lsp-mode.el

```diff
@@ -2973,6 +2973,7 @@ Return same value as `lsp--while-no-input' and respecting `non-essential'."
                              :error-handler (lambda (err) (setf resp-error err))
                              :mode 'detached
                              :cancel-token :sync-request)
+          (redisplay)
```

Comment:
Try wrapping it in `(when non-essential ...)`, too.

I'm not sure which is the best behavior, but waiting until the response is parsed before redisplay in the synchronous case makes sense to me.